src/ai/backend/manager/models/utils.py

```diff
@@ -4,6 +4,7 @@
 import functools
 import json
 import logging
+from contextlib import _AsyncGeneratorContextManager as AbstractAsyncCtxMgr
```

Comment:
You may use `contextlib.AbstractAsyncContextManager` as an abstract type instead of this concrete implementation class for type checks, though usually lsp like pyright/pylance suggests this one.